_posts/2016-09-21-approaching-tdd-outside-android.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+asset-type: post
+name: Approaching Outside-in TDD on Android I
+title: 'Approaching Outside-in TDD on Android I'
+author: Carlos Morera de la Chica
+image:
+    src: /assets/img/custom/blog/outside-in.png
+tags: 
+- android
+- tdd 
+- design
+canonical:
+    name: my personal blog
+    href: https://carlosmchica.github.io/approaching-tdd-outside-android/
+
+---
+
+<a href="http://martinfowler.com/articles/mocksArentStubs.html#DrivingTdd">Outside-in Test-Driven Development (TDD)</a> can be a challenge to implement. In this 3-part post series, we would like to share our experiences applying it to Android development and offer some practical tips for doing so yourself.
+
+As Android developers, we have been trying to apply the inside-out TDD style to our daily workflow but we thought that there was something missing.
+
+In this blog post series, we would like to share our experience applying Outside-in TDD to Android development. In this first post of the series we will introduce the necessary concepts and present our broad approach to the problem.
+
+Bearing in mind that we already have an architecture in our system, in most of the cases, we know the pieces that compose a feature's design upfront. That is why Outside-in TDD suits better our workflow. If you already know the design, outside-in let you go faster than the baby steps followed in the inside-out approach.
+
+We have been practising the outside-in workflow, again and again, and we would like to share how we have adapted the outside-in workflow to our platform, Android. In order to do so and to make it easier to follow, we have chosen the "Bank Kata" that Sandro uses on his screencast. We found it useful to show our approach to achieving the same effect on the Android platform, where the tools are different and it can be a bit tricky.
```

Comment:
repetition of the fact outside in is applied to the Android platform - maybe merge this with the paragraphs above? I propose to structure the post in paragraphs
- explain why outside-in is important
- explain why on Android
- describe what part I is about
- introduce example